### PR TITLE
 enforced that BOTH README template and properties file must exist for…

### PR DESCRIPTION
**Pull request by rhoads-zach**
… customized README to generate. Otherwise, will simply pass old README through as is.

### CODE REVIEW

**Comment by rhoads-zach** [comment]
@gastaldi mind reviewing?

**Comment by rhoads-zach** [comment]
@valdar mind trying my branch locally on your side?

File: addons/launcher-addon/src/main/java/io/fabric8/launcher/addon/ReadmeProcessor.java [hunk 1]
```diff
@@ -37,17 +40,23 @@ public String getReadmeTemplate(Mission mission) throws IOException {
...
-                props.load(url.openStream());
-        } catch (IOException io) {
-            // Do nothing
+        
+        URL url = getPropertiesURL(deploymentType.name().toLowerCase(), mission.getId(), runtime.getId());
+               
+        if (url != null) {
+            try (InputStream is = url.openStream()) {
+                props.load(is);
+                is.close();
```
Comment (gastaldi) [inline addons/launcher-addon/src/main/java/io/fabric8/launcher/addon/ReadmeProcessor.java]: No need to call close() here. The try block already does it

File: addons/launcher-addon/src/main/java/io/fabric8/launcher/addon/ui/booster/ProjectInfoStep.java [hunk 2]
```diff
@@ -293,7 +293,10 @@ public Result execute(UIExecutionContext context) throws Exception {
             }
         } catch (Exception e) {
             if (e instanceof FileNotFoundException) {
-                logger.log(Level.WARNING, "No README.adoc template found for " + mission.getId());
+                logger.log(Level.WARNING, "No README.adoc and properties found for " + mission.getId() + " " + runtime.getId() +
```
Comment (gastaldi) [inline addons/launcher-addon/src/main/java/io/fabric8/launcher/addon/ui/booster/ProjectInfoStep.java]: No big deal, but it would be interesting to have two catch clauses instead and log appropriately

Comment (gastaldi) [reply addons/launcher-addon/src/main/java/io/fabric8/launcher/addon/ui/booster/ProjectInfoStep.java]: You can also log e.getMessage() to know the file name that was not found

**Comment by rhoads-zach** [comment]
Thanks @gastaldi removed the close statement

File: README.md [hunk 3]
```diff
@@ -9,24 +9,24 @@ a Zip file containing a Maven project populated for an Eclipse Vert.x, Spring Bo
...
-$ cd web
-$ mvn wildfly-swarm:run
+$ java -jar web/target/launcher-backend-swarm.jar
 ```
 
-Then follow the [front-end README](https://github.com/fabric8-launch/launcher-frontend/blob/master/README.md) to run the front-end.
+* Set the environment variable for the frontend to use your local version of the backend
+    
+    $ export LAUNCHER_BACKEND_URL=http://localhost:8080/api/
+
```
Comment (quintesse) [inline README.md]: I don't think this information should be here, but in the frontend README.

Comment (quintesse) [reply README.md]: And it should then also mention the required:

    $ export LAUNCHER_MISSIONCONTROL_URL=ws://localhost:8080

(in the same frontend README)

**Comment by rhoads-zach** [comment]
@quintesse updated the Readme per feedback. Will file a separate PR for the front end readme

**Comment by rhoads-zach** [comment]
@quintesse updated front end readme here: https://github.com/fabric8-launcher/launcher-frontend/pull/220

**Comment by quintesse** [comment]
@rhoads-zach btw, as long as the title says "WIP, DO NOT MERGE" I'm not merging even though @gastaldi and I approved of the changes.

File: README.md [hunk 4]
```diff
@@ -9,21 +9,17 @@ a Zip file containing a Maven project populated for an Eclipse Vert.x, Spring Bo
...
-
 		$ export LAUNCHER_MISSIONCONTROL_SERVICE_HOST=localhost
 		$ export LAUNCHER_MISSIONCONTROL_SERVICE_PORT=8080
 
 * Execute the uber-jar in the target folder with:
-
 ```bash
-$ cd web
-$ mvn wildfly-swarm:run
+$ java -jar web/target/launcher-backend-swarm.jar
```
Comment (quintesse) [inline README.md]: Btw, the changes in this file should really have been in a separate PR as well. The fixes in this doc are totally unrelated to the code changes :)

**Comment by valdar** [comment]
Hi, I have just tested it locally and it seems to work: my project README.adoc file is preserved in the downloaded .zip.

**Comment by rhoads-zach** [comment]
thank you @valdar 

@quintesse @gastaldi I have removed the WIP as per @valdar comment since it works for him.

@quintesse would you like me to back out the other README change before you merge this?

File: README.md (hunk 4 above)
Comment (gastaldi) [reply README.md]: That is fine, we'll merge it anyway :+1:

**Comment by gastaldi** [comment]
Merged, thanks!

**Comment by quintesse** [comment]
> @quintesse would you like me to back out the other README change before you merge this?

No need @rhoads-zach , it was just a remark for future improvement :)
